database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -97,6 +97,7 @@ public void onChildMoved(DataSnapshot snapshot, String previousChildKey) {
         mSnapshots.remove(oldIndex);
         int newIndex = previousChildKey == null ? 0 : (getIndexForKey(previousChildKey) + 1);
         mSnapshots.add(newIndex, snapshot);
+        mSnapshotMap.put(snapshot.getKey(), newIndex);
```

Comment:
Note that the ordering implied by mSnapshots is not changed.  All that's really changed here is what getIndexForKey() does to fulfill its job.  There can be only one item with a particular key here, and that relationship is simply be expressed in the map now, so it doesn't have to scan the list for the unique key.

There is a test that checks for order changes called testChangePriorities().  It alters the priority of an item so that the array order changes from 1,2,3 to 3,1,2.  This test passes.